Source.cpp

```diff
@@ -0,0 +1,47 @@
+#include <iostream>
+
+using std::cout;
+using std::endl;
+
+int fib(int num){
+
+	int nb4 = 1;
+	int n = 0;
+	int value = 1;
+
+	if (num == 1)
+		return 1;
+	if (num == 2)
+		return 1;
+	else
+	{
+		for (int i = 1; i <= num; i++){
+
+			value = nb4 + n;
+			nb4 = n;
+			n = value;
+		}
+	}
+
+	return value;
+}
+
+int recfib(int num, int f1 = 1, int f2= 1){
+
+	int sum = 0;
+
+	for (int y = 0; y < num; y++){
+		sum = f1 + f2;
+		recfib(num--, f1 = f2, f2 = sum);
+	}
+
+	return sum;
+}
```

Comment:
this is not really a recursive function, since the return value of the recursive call is thrown away.  it looks really close to a good iterative solution, but it doesn't look like it'll work (did you test it?).  the recursive call is almost right though :-).  you should probably avoid the `--` and `++` operators inside recursive calls though: it looks like `num` might not be decremented until after the call, though i'd have to check to make sure.  my recursive code, for reference:
```
int fibonacci(int n, int f1 = 1, int f2 = 1) {
    // error check
    if (n < 1)
        return 0;

    if (n == 1)
        return f1;
    if (n == 2)
        return f2;

    return fibonacci(n-1, f2, f1+f2);
}
```